llvm/lib/IR/DiagnosticInfo.cpp

```diff
@@ -157,8 +157,10 @@ std::string DiagnosticInfoWithLocationBase::getLocationStr() const {
   StringRef Filename("<unknown>");
   unsigned Line = 0;
   unsigned Column = 0;
-  if (isLocationAvailable())
+  if (isLocationAvailable()) {
     getLocation(Filename, Line, Column);
+    return (getAbsolutePath() + ":" + Twine(Line) + ":" + Twine(Column)).str();
```

Comment:
Couldn't you do this at the point you display the diagnostic?

Reply:
This is calling getLocation, then discarding the filename? Shouldn't this also be up to the client's decision on how to display the path?